beets/dbcore/query.py

```diff
@@ -550,19 +551,38 @@ def parse(cls, string):
         """Parse a date and return a `Period` object or `None` if the
         string is empty.
         """
-        if not string:
-            return None
-        ordinal = string.count('-')
-        if ordinal >= len(cls.date_formats):
-            # Too many components.
-            return None
-        date_format = cls.date_formats[ordinal]
-        try:
-            date = datetime.strptime(string, date_format)
-        except ValueError:
-            # Parsing failed.
-            return None
-        precision = cls.precisions[ordinal]
+        pattern_dq = '@([+|-]?)(\d+)([y|m|w|d])'
```

Comment:
Some comments here (and below) would be *really* helpful. Specifically, at this point, it would be great to note that we are making a decision about whether the period seems to be relative or absolute.